api/user.go

```diff
@@ -923,6 +925,40 @@ func getUser(c *Context, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getProfile(c *Context, w http.ResponseWriter, r *http.Request) {
+	params := mux.Vars(r)
+	id := params["id"]
+
+	if result := <-Srv.Store.User().Get(id); result.Err != nil {
+		c.Err = result.Err
+		return
+	} else if HandleEtag(result.Data.(*model.User).Etag(), w, r) {
```

Comment:
I think we want to check permissions before handling the etag. This way if a user permissions change they get updated info.